### ISSUE TITLE
provide support for faker.js 3.1.0 version

### ISSUE THREAD

**Issue by ducin** [json-schema-faker/json-schema-faker#117]
After Marak has bumped up faker version from 3.0.1 to 3.1.0, json-schema-faker broke because of semver set to ^3.0.1. This has been changed to ~3.0.1 since (#116). Now the task is to provide the support for `faker@3.1.0`.

Try to install `json-schema-faker` where faker is `3.1.0` (the latest) and paste following code:
```
var jsf = require('json-schema-faker');
var schema = {
    "type": "string",
    "faker": "random.uuid"
};

jsf(schema);
```

You'll get this error:
```
c:\Users\tomasz.ducin\Development\Tests\jsftest\node_modules\json-schema-faker\lib\jsf.js:74
      throw new Error(e.message + ' in ' + '/' + e.path.join('/'));
      ^

Error: self.number is not a function in /
    at generate (c:\Users\tomasz.ducin\Development\Tests\jsftest\node_modules\json-schema-faker\lib\jsf.js:74:13)
    at Object.<anonymous> (c:\Users\tomasz.ducin\Development\Tests\jsftest\test.js:7:1)
    at Module._compile (module.js:435:26)
    at Object.Module._extensions..js (module.js:442:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:313:12)
    at Module.runMain [as _onTimeout] (module.js:467:10)
    at Timer.listOnTimeout (timers.js:92:15)
```

Actually, the exception thrown is not on our code, it's in [random.js file inside faker](https://github.com/Marak/faker.js/blob/649843ade73b730911506b8ecc6bf64bad10238c/lib/random.js#L90-L97). I believe faker.js has some internals modified (the way the modules are being built) and our jsf assumed the something else that is not here anymore, hence the exception. This is the [exact failing line](https://github.com/Marak/faker.js/blob/649843ade73b730911506b8ecc6bf64bad10238c/lib/random.js#L93) but if you try to run the `random.uuid` faker generator, it works perfectly fine.

Need to investigate it.

**Comment by ducin** [comment]
Faker.js has changed the `random.uuid` implementation from:

```
  this.uuid = function () {
      var RFC4122_TEMPLATE = 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx';
      var replacePlaceholders = function (placeholder) {
          var random = Math.random()*16|0;
          var value = placeholder == 'x' ? random : (random &0x3 | 0x8);
          return value.toString(16);
      };
      return RFC4122_TEMPLATE.replace(/[xy]/g, replacePlaceholders);
  }
```

to 

```
  this.uuid = function () {
      var self = this;
      var RFC4122_TEMPLATE = 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx';
      var replacePlaceholders = function (placeholder) {
          var random = self.number({ min: 0, max: 15 });
          var value = placeholder == 'x' ? random : (random &0x3 | 0x8);
          return value.toString(16);
      };
      return RFC4122_TEMPLATE.replace(/[xy]/g, replacePlaceholders);
  }
```

and that;t the reason  for the break. Before, there was no `this`. Now, this is overwritten from jsf's `lib/string.js` file.

**Comment by ducin** [comment]
Seems like theproblem is solved in `v0.2.15` in 1a9a7ffcdd875306fc83fa789f7f153a166b2a96
